src/main/java/racingcar/domain/Cars.java

```diff
@@ -1,50 +1,37 @@
 package racingcar.domain;
 
-import java.util.ArrayList;
+import static java.util.stream.Collectors.*;
+
+import java.util.Arrays;
 import java.util.List;
 
 public class Cars {
-    public static final int INITIAL_POSITION = 0;
-
     private final List<Car> cars;
 
-    public Cars(String[] names) {
-        this.cars = new ArrayList<>();
-        for (String name : names) {
-            cars.add(new Car(name, INITIAL_POSITION));
-        }
+    private Cars(List<Car> cars) {
+        this.cars = cars;
     }
 
-    public void moveAll() {
+    public static Cars of(String[] names) {
+        return new Cars(Arrays.stream(names).map(Car::new).collect(toList()));
+    }
+
+    public void moveAll(MovingStrategy movingStrategy) {
         for (Car car : cars) {
-            car.move();
+            car.move(movingStrategy);
         }
     }
 
     public List<String> getChampions() {
         return findChampions(getHighScore());
     }
 
-    private int getHighScore() {
-        int highScore = 0;
-        for (Car car : cars) {
-            highScore = Math.max(highScore, car.getPosition());
-        }
-        return highScore;
-    }
-
     private List<String> findChampions(int highScore) {
-        List<String> champions = new ArrayList<>();
-        for (Car car : cars) {
-            verifyChampion(champions, car, highScore);
-        }
-        return champions;
+        return cars.stream().filter(car -> car.isSamePosition(highScore)).map(Car::getName).collect(toList());
```

Comment:
스트림을 사용할 경우에 `.`를 기준으로 적절히 개행을 해준다면 보다 가독성 좋은 코드를 만들 수 있어요!! 
해당 코드는 제가 샘플을 보여드릴게요 :) 아래 동일한 부분에 대해서 쿼리치가 한번 개선해보면 좋을 것 같아요!!

```suggestion
        return cars.stream()
                .filter(car -> car.isSamePosition(highScore))
                .map(Car::getName)
                .collect(toList());
```